runtime/doc/vim9.txt

```diff
@@ -1567,352 +1606,847 @@ If the return type is "void" the function does not return a value.
+the former, so, this is okay: >vim9
+
+	vim9cmd echo [1, 2]->extend(['3'])	# [1, 2, 3]
+<
+whereas, this is not: >vim9
+
+	vim9script
+	var mylist: list<number> = [1, 2]
+	echo mylist->extend(['3'])	   # E1013: Argument 2: type mismatch
+<
+Using |extendnew()| is needed for extending an existing typed list, except
+where the extension matches the list's type (or it is "any").  For example,
+first extending with an element of the same type, then extending with a
+different type: >vim9
+
+	vim9script
+	var mylist: list<number> = [1, 2]
+	mylist->extend([3])
+	echo mylist->extendnew(['4'])	   # [1, 2, 3, '4']
+
+<								*E1158*
+Using |flatten()| is not allowed in Vim9 script, because it is intended
+always to change the type.  This even applies to a list literal
+(unlike |map()| and |extend()|).  Instead, use |flattennew()|: >vim9
+
+	vim9cmd [1, [2, 3]]->flatten()		# E1158: Cannot use flatten
+	vim9cmd echo [1, [2, 3]]->flattennew()	# [1, 2, 3]
 
 Assigning to a funcref with specified arguments (see |vim9-func-declaration|)
-does strict type checking of the arguments. For variable number of arguments
-the type must match: >
-	var FuncRef: func(string, number, bool): number
-	FuncRef = (v1: string, v2: number, v3: bool) => 777	# OK
-	FuncRef = (v1: string, v2: number, v3: number) => 777	# Error!
-	# variable number of arguments must have same type
-	var FuncVA: func(...list<string>): number
-	FuncVA = (...v: list<number>): number => v  # Error!
-	FuncVA = (...v: list<any>): number => v	    # OK, `any` runtime check
-	FuncVA = (v1: string, v: string2): number => 333     # Error!
-	FuncVA = (v: list<string>): number => 3	    # Error!
-
-If the destination funcref has no specified arguments, then there is no
-argument type checking: >
-	var FuncUnknownArgs: func: number
-	FuncUnknownArgs = (v): number => v			# OK
-	FuncUnknownArgs = (v1: string, v2: string): number => 3	# OK
-	FuncUnknownArgs = (...v1: list<string>): number => 333	# OK
-<
-			 *E1211* *E1217* *E1218* *E1219* *E1220* *E1221*
-			 *E1222* *E1223* *E1224* *E1225* *E1226* *E1227*
-			 *E1228* *E1235* *E1238* *E1250* *E1251* *E1252*
-			 *E1253* *E1256* *E1297* *E1298* *E1301* *E1528*
-			 *E1529* *E1530* *E1531* *E1534*
+involves strict type checking of the arguments.  For example, this works: >vim9
+
+	vim9script
+	var F_name_age: func(string, number): string
+	F_name_age = (n: string, a: number): string => $"Name: {n}, Age: {a}"
+	echo F_name_age('Bob', 42)
+<
+whereas this fails with error |E1012| (type mismatch): >vim9
+
+	vim9script
+	var F_name_age: func(string, number): string
+	F_name_age = (n: string, a: string): string => $"Name: {n}, Age: {a}"
+
+If there is a variable number of arguments they must have the same type, as in
+this example: >vim9
+
+	vim9script
+	var Fproduct: func(...list<number>): number
+	Fproduct = (...v: list<number>): number => reduce(v, (a, b) => a * b)
+	echo Fproduct(3, 2, 4)  # Echoes 24
+
+Though, <any> may be used to accommodate mixed types: >vim9
+
```

Comment:
At original text line 1747, there is
> FuncVA = (...v: list<any>): number => v	    # OK, `any` runtime check

Maybe an example that compiles OK, but gets a runtime error would help to convey the compile/runtime difference.

Reply:
Agreed. I think we're saying the same thing, though your example specifically demonstrates how `any` "messes with the type system", postponing checking to runtime.  However, my question was whether:

- it's worth noting it is possible, though not a good idea / discourage it, or
- omit it entirely.

Your example demonstrates the runtime failure clearly whereas I'd only commented/hinted to it. To show the pitfall, a modified version of mine works:

<img width="837" height="208" alt="image" src="https://github.com/user-attachments/assets/032288aa-70d2-49f9-9831-6a645ec03d57" />